### DISCUSSION TITLE
LITIGATION npl_class: Parsing and Consolidation ?

### DISCUSSION THREAD

**Discussion by cverluise** [cverluise/PatCit#23] [Projects & Ideas]
Hello,

Since v0.2-npl, we have introduced a LITIGATION `npl_class`

### Question
How should this class of citations be structured and consolidated?

### Food for thoughts
Below is a sample of 100 citations for this class.

<details>

```sql
WITH
  tmp AS (
  SELECT
    npl_publn_id
  FROM
    `npl-parsing.patcit.v02_npl`
  WHERE
    npl_class="LITIGATION")
SELECT
  npl_biblio
FROM
  `usptobias.patstat.tls214` AS tls214,
  tmp
WHERE
  tmp.npl_publn_id = tls214.npl_publn_id
LIMIT 100 
```

  |npl_biblio |  
-- | --
1 | Opposition filed by Sanofi-Aventis on Jan. 27, 2010 for EP 1732949. |  
2 | Child identification card, 1995. |  
3 | CHIA SHEN ET AL: 'RESOURCE RECLAIMING IN MULTIPROCESSOR REAL-TIME SYSTEMS', IEEE TRANSACTIONS ON PARALLEL AND DISTRIBUTED SYSTEMS, IEEE INC, NEW YORK, US, vol. 4, no. 4, 1 April 1993 (1993-04-01), pages 382 - 397, XP000381810, ISSN: 1045-9219 |  
4 | Suggestion for Interference filed in connection with U.S. Appl. No. 10/689,866 on May 11, 2010. |  
5 | J. A. Nelder et al., A Simplex Method for Function Minimization, Computer Journal , 7, 308 313 (1965). |  
6 | IPR2014-00946 Decision Institute of Inter Partes Review dated Dec. 18, 2014. |  
7 | AirSep, A Chart Industries Company, PSA Oxygen Generator AS-A,HVLPO2, LLC v Oxygen Frog, LLC et al. United States District Court for the District of Nebraska, Defendant's Motion for Transfer of Venue, Case No. 8:15-cv-00176-JMG-CRZ, Doc # 10, pp. 7-8, Publication date unknown, Filed Jun. 4, 2015, Printed Mar. 30, 2016. |  
8 | First Amended Complaint for Patent Infringement, Server Technology, Inc. v. American Power Conversion Corporation, Case No. 3:06-CV-00698-LRH-VPC, 83 pp. (Feb. 20, 2007). |  
9 | v. 3-Dimensional Pharmaceuticals, U.S. District Court, Delaware, Jan. 7, 2000. |  
10 | David Stirrup; 'Workflow Management Coalition defines Interface between Workflow Engines and Process Definition Information'; Internet; http://www.aiim.org/wfmc; downloaded Jan. 1999; all. |  
11 | French Application Registration No. 96 13705 filed on May 11, 1996 entitled 'Anti-Fraud System for Fluid Distribution Network' in its entirety, 20 pages. |  
12 | 'Complaint', Waters Industries, Inc. v. Mr. Christmas Incorporated, et al., United States District Court for the Northern District of Illinois, Case No. 1:09-cv-07577, 38 pages. |  
13 | Counterclaim-Defendants' Amended Invalidity Contentions Pursuant to Patent Rule 3-3, signed by James A. Jorgensen, Feb. 8, 2008, Filed in Civil Action No. 6:06-CV-222 (LED) in the United Stated District Court for the Eastern District of Texas, Tyler Division, 19 pages (nonmaterial portions redacted). |  
14 | ThinkVillage-Kerfoot LLC v. Groundwater & Environmental Services, Inc., Plaintiff's Response to Defendant Groundwater & Environmental Services, Inc.'s Amended Counterclaims, Civil Action No. 1:08-cv-11711-GAO, Dec. 30, 2008, 5 pages. |  
15 | Exhibit B (Proposed Constructions for Identified Terms) to Amended Joint Claim Construction Chart and Statement, in the matter of Life Technologies Corp. v. Illumina, Inc., 09-706 RK, U.S. District Court for the District of Delaware. |  
16 | Ralph Wilhelm, Jr., Ph.D., Declaration of Ralph Wilhelm, Jr., Ph.D., Inter Partes Review of U.S. Pat. No. 7,650,210, IPR2013-00415, Jul. 7, 2013. |  
17 | Declaration of Magnus Hägglund, Department Head Fibres Production Pulp, M-real Sweden AB, Husum-Wifsta Fabriker, dated Jun. 13, 2006. |  
18 | Scientific Software, Inc.'s Response to NuGenesis First Set of Document Requests dated Nov. 12, 2001. |  
19 | Exhibit B11 to Defendants' Preliminary Invalidity Contentions-Invalidity Chart for U.S. Patent No. 7,570,954 based on U.S. Patent No. 4,736,371 ('Tejima'). |  
20 | European publication No. 0 317 325 A2 cover sheet; specification p. 5; Figs. 1,2, May 24, 1989. |  
21 | Portions of Prosecution history of AU2012340387, Dec. 3, 2015, Nicira, Inc. |  
22 | Notice of Panel Decision from Pre-Appeal Brief Review, issued by the United States Patent and Trademark Office in connection with U.S. Appl. No. 12/135,066, on May 31, 2011, 2 pages. |  
23 | Opposition No. 2001-71700-01 against Japanese Patent No. 3117966. |  
24 | Statutory Declaration by Stuart Graham Tangye dated Jun. 28, 2011. |  
25 | Declaration of Daniel J. Carr. |  
26 | Grant, Matthew, Quick Start for Beginners to Drive a Stepper Motor, Freescale Semiconductor, Rev. 1, 06/2005, Pages 1, 5. |  
27 | Motorola, Inc. R1 , Motorola, Inc., 1990. |  
28 | Rossin Opposition 2 Errata, Patent Interference No. 105,402 (McK), Technology Center 1700, Jul. 19, 2006. |  
29 | Examiner Annotated Figure 4 of Arai |  
30 | Samsung, LG and HTC, Petitioner v. Russell W. White and Kevin R. Imes, Patent Owner, “Corrected Petition for Inter Partes Review of U.S. Pat. No. 7,953,390*,” Case IPR2014-00212, U.S. Pat. No. 7,953,390, Dec. 12, 2013, 67 pages. |  
31 | Petition for Inter Partes Review for U.S. Patent No. 8,079,413 (U.S. Appl. No. 13/194,871); Case No. 2013-00231; Filed Apr. 2, 2013; Administrative Patent Judge Sally C. Medley. |  
32 | Declaration of Dr. Steven M. Nesbit, U.S. Pat. No. 8,905,855, Feb. 24, 2016, 235 Pages. |  
33 | EX 2050, Brown et al. v. Schrage et al., dated Dec. 25, 2011, Patent Interference No. 105,799, United States Patent and Trademark Office, Board of Patent Appeals and Interferences. |  
34 | Declaration of Dr. Stephen H. Smith, in Support of Petition for Inter Partes Review (IPR No. 2015-00786), Wright Medical Technology, Inc., Petitioner, v. BioMedical Enterprises, Inc. Patent Owner, U.S. Pat. No. 8,584,853 to Knight et al., Filed Feb. 20, 2015. |  
35 | Synopsis on Set Public Hearing Nov. 14, 1997 to Adopt Proposed Rule 1138-'Control of Emissions from Restaurant Operations', South Coast Air Quality Management District, Agenda No. 5. |  
36 | Esterification and/or Ester Interchange Catalyst; Industrial Organic Chemicals, Leather, Fats and Waxes, Date Unknown. |  
37 | Applicant s Exhibit A Fasson Graphic Arts Division, 250 Chester Street, Painesville, OH 44077; Fasson Roll Printer News , Fall 1975; 10 pages. |  
38 | Patent Owners Request for Oral Argument in IPR2015-00266, filed on Nov. 4, 2015 (Paper No. 38). |  
39 | 'Micro Motion, Inc.'s Opposition to Invensys Systems, Inc.'s Motion to Dismiss and Strike Micro Motion's Counterclaim and Affirmative Defense of Inequitable Conduct', filed Oct. 25, 2013, Case No. 6:12-00799, related to U.S. Patent Nos. 6,311,136; 6,754,594; 7,124,646; 7,136,761; 7,505,854; 7,571,062 and 8,000,906, 185 pgs. (Part 2 of 2). |  
40 | Wustlich Mikro-/Opto-Elektronik GmbH (1994/1995). |  
41 | Opposition No. 2001-73554 against Japanese Patent No. 3187400. |  
42 | 95/001,566 RX—Mail Petition Decision—Dismissed Jan. 26, 2012. |  
43 | Rodriguez Individual to tap Internet with an agent based new service , InfoWorld v16 n43 p. 58, Oct. 24, 1994. |  
44 | Defendant Emulex Corporation's Disclosure of Preliminary Invalidity Contentions and Exhibit F, Broadcom v. Emulex Corporation, Case No. SACV 09-1058-JVS (ANx), dated Jun. 28, 2010. |  
45 | U.S. Federal Highway Administration (FHWA), Proceedings , International Conference on Design and Construction of Deep Foundations, Dec. 1994. |  
46 | 1st Petition for Inter Partes Review of U.S. Patent No. D631,670 (68 pages, filed Jun. 19, 2015 by Petitioners Johns Manville Corporation and Johns Manville, Inc.). |  
47 | Defendants Invalidity Contentions, ExhibitC06, Ultravision Technologies v. Lamar et al., E.D. Texas, Case No. 2:16-cv-374, Sep. 9, 2016, pp. 1-79. |  
48 | District Court Civil Docket No. 17-1: Declaration of Jeffrey C. Brown regarding (16) First Motion to Expedite Discovery and Supporting Brief, Entered: Jan. 25, 2005. |  
49 | ITU-T Recommendation Q. 765 (1998), Signaling system No. 7-Application transport mechanism. |  
50 | Likewise Software, Inc.'s Answer, Affirmative Defenses and Counterclaims, Quest Software, Inc. v. Centrify Corporation and Likewise Software, Inc., Case No. 2:10-CV-00859-TS, United States District Court for the District of Utah, Central Division. |  
51 | W.C. CHAN AND P.D. WHITE,: 'Fmoc Solid Phase Peptide Synthesis', 2000, OXFORD UNIVERSITY PRESS |  
52 | Rothfuss, James. 'Windows NT 4.0 Workstation Security Advisor, version 1.3', May 1998. |  
53 | Medline Industries, Inc. vs C.R. Bard, Inc; No. 1:14-cv-03618; Response to Medline Industries, Inc.'s First Set of Interrogatories; Dated Sep. 26, 2014. |  
54 | United States District Court Central District of California, Order Granting Defendants' Motion for Summary Judgment, Jul. 20, 2006, pp. 1-77, USA. |  
55 | In re Inter Partes Review of USP 8,286,561, Second Declaration of Bradley M. Nall, Trial No. IPR2013-00358 (Feb. 6, 2014). |  
56 | HTC Europe Co. Ltd and Apple Inc. Invalidity Claim Dated: Jul. 29, 2011, Together with Amended Particulars of Claim and Amended Grounds of Invalidity, 22 pages. |  
57 | Cooled Tip Ablation Results in Increased Radiofrequency Power Delivery and Lesion Size, PACE, vol. 17, Apr. 1994, Part II, p. 782. |  
58 | Order-Bd.R. 104(c) for Patent Interference No. 105,753 dated Aug. 11, 2010; 3 pages. |  
59 | Review Article Insensitive Adiabatic RF Pulses (39 pages). |  
60 | Exhibit F-3 of 'Initial Disclosure of Noninfringement, Invalidity, and Unenforceability Contentions'; Case No. 3:13-CV-00579-RCJ-WGC; Unwired Planet vs. Square, Inc.; Apr. 22, 2014. |  
61 | BPAI Interference No. 105,753, Omura v. Shafer, Omura Response to Order to Show Cause, filed Jul. 20, 2010. |  
62 | Complaint and Demand for Jury Trial, 3M Innovative Properties Company and 3M Company vs. Louis M. Gerson Co, Inc., and Gerson Professional Products, Inc., Civil No. 08-04960 JRT-FLN (U.S. District Court, District of Minnesota), filed Aug. 19, 2008 (30 pages). |  
63 | Third Party Requester's Opposition to Petition to Vacate, for Reexamination Control No. 95/001,806, filed Mar. 1, 2012, 15 pages. |  
64 | Appendix B of IBM's Invalidity Contentions dated Apr. 27, 2007, in matter of IBM v. Amazon.com, Inc. (case 6:06-cv-452). |  
65 | IPR2015-01517, Patent Owner's Preliminary Response, dated Oct. 19, 2015 (70 Pages). |  
66 | Statutory Declaration of Garth Stanley Anderson and attached exhibits. |  
67 | European Telecommunications Standard Institute: ETSI TS 102613, Smart Cards; UICC-Contactless Front-end (CLF) Interface; Part 1: Physical and Data Link Layer Characteristics, pp. 1-58, Seventh Edition, Reference No. ETSI TS 102613 v.7.6.0, ETSI TS (2009-2010). |  
68 | GTITM UPOVROM Citation for 'Celina' as per QZ PBR 991029; Jul. 19, 1999. |  
69 | Notice of Opposition to a European Patent (EP 1 811 013) filed by 3M Innovative Properties Company against Gojo Industries dated May 12, 2010. |  
70 | Nikolaos Papanikolopoulos, Ph.D., Declaration of Nikolaos Papanikolopoulos, Ph.D., Inter Partes Review of U.S. Patent No. 6,772,057, IPR2013-00419, Jul. 12, 2013. |  
71 | LifeShield TKO Anti-Reflux Device Brochure dated Feb. 8, 2008. |  
72 | Samtec Brochures, some dated Jul. 2001, some dated May 2000, some dated Dec. 1999, and other undated (85 pages total). |  
73 | Opposition Statement for European Patent Application No. EP 1 717 061, English translation, dated Sep. 21, 2010, 12 pages. |  
74 | Communication from Opposition Division Revoking the EP Patent EP-B-979059, dated Apr. 15, 2008, Apr. 24, 2008 and May 29, 2008. |  
75 | Updated portions of prosecution history of U.S. Appl. No. 11/129,134. Dec. 30, 2010, Gallagher, Michael D., et al. |  
76 | Exhibit D2, Asserted Claims and Infringement Contentions by Plaintiff VirnetX, Inc. against Apple based on U.S. Pat. No. 7,921,211 Patent. |  
77 | AngioScore, Inc. v. Trireme Medical, LLC, Verdict Form ftiled Sep. 29, 2015, United States District Court, Northern District of California, Oakland Division, Case No. 4:12-cv-3393-YGR. |  
78 | European Opposition document for EP Patent No. 1 012 274-Minutes from the oral proceedings dated Jun. 22, 2010. |  
79 | DriveCam, Inc.'s Infringement Contentions Exhibit C, U.S. Patent 7,804,426. Aug. 19, 2011. |  
80 | ITC Investigation of Certain Products Containing Interactive Program Guide and Parental Control Technology, Investigation No. 337-TA-845, 'Complainants' Petition for Review of Final Initial Determination' Jun. 24, 2013 (55 pages). |  
81 | sequence alignment of SEQ ID NO 23 with UniProt access no C9EJF3_PCV2 November 2009 |  
82 | 6:12-cv-17, -20, -120, Memorandum Opinion and Order, U.S. District Court for the Eastern District of Texas, U.S. Magistrate Judge Caroline M. Craven, Mar. 12, 2014, 34 pgs. |  
83 | Sep. 21, 2011 print of VEMAG Casing Holding Device Type 836 order confirmation (Atameh Pars Teyhoo Co.), purportedly dated Jul. 11, 2005. |  
84 | Answer to Counterclaim (May 4, 2009) U.S. District Court, Northern District of California Civil Action No. CV 09-0982. |  
85 | Voluntary Amendments filed Apr. 15, 2014 in EP 12778483.3. |  
86 | Opinion and Order, Motion for Reconsideration of the Court's Claim Construction Ruling, Knowles Electronics, LLC v. Analog Devices, Inc., United States District Court for the Northern District of Illinois, Civil Action No. 1:11-cv-06804 (Feb. 19, 2013). |  
87 | Marineland Commercial Aquariums Model LB 12BW, dated prior to filing date of present application but after Mar. 1994. |  
88 | Exhibit B—Netflix, Inc. v. Rovi Corp. et al. (4:11-cv-06591, N.D. Cal.), Netflix's Disclosure of Invalidity Contentions under Patent L.R. 3-3 dated Oct. 31, 2014, Invalidity of Claims 1, 67, 12-13 and 17 of U.S. Pat. No. 6,898,762 Under 35 U.S.C. § 103 (“Primary References”), 96 pages. |  
89 | RFC 3135-Performance Enhancing Proxies Intended to Mitigate Link-Related Degradations, available online at: http://www.faqs.org/rfcs/rfc3135.html. |  
90 | Declaration of Joseph C. McAlexander, III in the Civil Action No. 01-CV-4435, Dec. 15, 2004. |  
91 | Opinion and Order by the Court; Learning Curve Brands, Inc. v. Munchkin, Inc.; Case No. 09-cv-416-bbc; Mar. 23, 2010. |  
92 | Letter from Mr. Gerald Knapp of CableData to Rodney A. Hanson of Westar Communications, Inc., bearing a date of Mar. 19, 1985 (Plaintiff's Exhibit 325). |  
93 | Abbott's Third Supplemental Answers to Innogenetics' First Set of Interrogatories U.S. District Court for the Western District of Wisconsin, Case No. 05C 0575 C, Filed Sep. 29, 2005. |  
94 | Plaintiffs' Pretrial Memorandum filed by Emory University, Gilead Sciences, Inc.,Case No. 1:08-cv-10838-RJS (Filed Sep. 9, 2013), 60 pages. |  
95 | Updated portions of prosecution history of EP07107470, Apr. 19, 2011, Apple Inc. |  
96 | Defendants Exhibit 47 (DDX 47), cited in Mona Darwish Deposition Transcript. |  
97 | STIC sequence search performed for sequences in parent application 13/203934 on 5/2013. Pages 1-654. |  
98 | Jury Verdict, dated Jul. 1, 2005 (Case No. 03-027-SLR and Case No. 03-283-SLR). |  
99 | 'Responsive Claim Construction and Identification of Extrinsic Evidence of Defendani/Counterclaimant SmartDrive Systems, Inc.' Claim Construction and and Extrinsic Evidence in DriveCam, Inc. v. SmartDrive Systems, Inc., Case No. 3:11-CV-00997-H (RBB), for the Southern District of California. Nov. 15, 2011, 20 pages. |  
100 | Memorandum in Support of Defendant\s Motion to Dismiss Second Amended Complaint Due to Patent Invalidity Under 35 U.S.C. § 101; NDIL Case 14-cv-05197; Exhibit B; U.S. Pat. No. 7,852,212; Mar. 17, 2015.

</details>

### Open questions

- Need for further sub-classification? If yes, which sub-classes?
- Fields to be parsed?
- Corpus/ reference database for consolidation?

I leave these questions open to discussion.

Cheers

**Comment by gderasse** [comment]
Did you have a look at patents-public-data:uspto_oce_litigation on BigQuery? It could provide useful food for thought.

**Comment by cverluise** [comment]
Thanks @gderasse.

Some more resources:

- [Patent Litigation Data from US District Court Electronic Records (1963-2015)](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=2942295): original paper describing the dataset (1963-2015) Nb: updated to include data up to 2019 since that time
- [USPTO Patent Number and Case Code File Dataset Documentation](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3507607): document describing the hand-coded patent and case type variables
- [USPTO Patent litigation docket reports data](https://www.uspto.gov/learning-and-resources/electronic-data-products/patent-litigation-docket-reports-data)

From what I can see, this could be helpful to shape the parsing and consolidation (PACER) strategies for US litigations. 2 limitations still:

- rest of the world
- IPR, petititions etc don't seem to be covered (I might have missed it)

**Comment by cverluise** [comment]
*Follow-up on the IPR limitation*: IPR are part of PTAB and are accessible through:

- the [PTAB open data (bulk)](https://developer.uspto.gov/ptab-web/#/search/decisions)
- and the related [API](https://developer.uspto.gov/api-catalog/ptab-api-v2)

**Comment by cverluise** [comment]
At least fo english legal NPL citations, we should experiment with this exciting [Blackstone project](https://github.com/ICLRandD/Blackstone#named-entity-recogniser)

**Comment by cverluise** [comment]
The v2 of blackstone should be released in a couple of weeks. Contributors have already announced substantial improvements (in particular for the NER). Let's wait until that release

**Comment by cverluise** [comment]
Might be interesting to have a look at that even though the extraction part seems to have been externalized: https://case.law/about/

**Comment by gderasse** [comment]
Definitely! They make the data available in bulk for research. We could use the information either for entity identification/disambiguation or for interoperability if they offer stable ids.
